src/forms/Textarea.jsx

```diff
@@ -113,82 +111,47 @@ export class Textarea extends React.PureComponent<Props, State> {
 			name,
 			value, // eslint-disable-line no-unused-vars
 			label,
-			labelClassName,
-			className,
-			error,
+			labelClassName, // eslint-disable-line no-unused-vars
 			rows,
 			style = {},
 			maxHeight,
 			minHeight,
 			maxLength,
 			id,
 			onChange, // eslint-disable-line no-unused-vars
-			autosize,
 			helperText,
 			required,
-			requiredText,
 			disableResize,
 			...other
 		} = this.props;
 
-		const classNames = {
-			textarea: cx(
-				'span--100',
-				{
-					'field--error': error,
-					'textarea--autoheight': autosize,
-					'textarea-no-resize': disableResize,
-				},
-				className
-			),
-			label: cx(
-				'label--field',
-				{
-					'label--required': required,
-					'flush--bottom': helperText,
-				},
-				labelClassName
-			),
-			helperText: cx('helperTextContainer', { required }),
-		};
-
 		// Character limits should be a "soft" limit.
 		// Avoid passing maxLength as an HTML attribute
 		if (maxLength) delete other.maxLength;
 
 		return (
 			<div className="inputContainer">
-				{label && (
-					<label
-						className={classNames.label}
-						htmlFor={id}
-						data-requiredtext={required && requiredText}
-					>
-						{label}
-					</label>
-				)}
-				{helperText && <div className={classNames.helperText}>{helperText}</div>}
-				<textarea
+				{label && <FieldLabel htmlFor={id}>{label}</FieldLabel>}
+				{helperText && <FieldHelper>{helperText}</FieldHelper>}
+				<SwarmTextarea
 					type="text"
 					name={name}
 					required={required}
-					className={classNames.textarea}
 					onChange={this.onChange}
 					rows={rows}
 					ref={textarea => {
 						this.textarea = textarea;
 					}}
-					style={{ minHeight, maxHeight, ...style }}
+					style={{
+						minHeight,
+						maxHeight,
+						resize: disableResize ? 'none' : 'auto',
+						...style,
+					}}
 					id={id}
 					value={this.state.value}
 					{...other}
 				/>
-				{maxLength && (
-					<CharCounter
-						maxLength={parseInt(maxLength, 10)}
-						valueLength={parseInt(this.state.value.length, 10)}
-					/>
-				)}
```

Comment:
I remember you mentioned this, we don't have to support this for the first release either?